nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-json-record-utils/src/main/java/org/apache/nifi/json/AbstractJsonRowRecordReader.java

```diff
@@ -122,7 +156,15 @@ protected AbstractJsonRowRecordReader(final InputStream in,
         capturedFields = new LinkedHashMap<>();
 
         try {
-            jsonParser = jsonFactory.createParser(in);
+            final ObjectMapper codec = new ObjectMapper();
+            if (allowComments) {
+                codec.enable(JsonParser.Feature.ALLOW_COMMENTS);
+            }
+            if (streamReadConstraints != null) {
+                codec.getFactory().setStreamReadConstraints(streamReadConstraints);
+            }
```

Comment:
Instead of allowing this to be null, it would be more consistent to to always create a default instance of the constraints, and set 20 MB as the value, aligning with current Jackson 2.15 defaults.